File: 2020/day1/day1_p1.kt
// my first Kotlin!
// let the lessons come from actually doing the exercises

import java.io.File

fun main() {
	val fileName = "input.txt"
	val lines: List<String> = File(fileName).readLines()
	//
	val nums = lines.map { line -> line.toInt() }

	for ((index, num) in nums.withIndex()) {
		
		for (other in nums.slice(index..nums.size-1)) {
			if (num + other == 2020) {
				println(num*other)
			}
		}
	}
}

File: 2020/day1/day1_p1.py

with open('input.txt', 'r') as tf:
	nums = [int(l.strip()) for l in tf.readlines()]

for ind in range(len(nums)):
	curr_num = nums[ind]

	for other in nums[ind+1:]:
		sum = curr_num + other
		if sum == 2020:
			print(f'{curr_num} + {other} = 2020')
			print(f'{curr_num} * {other} = {curr_num*other}')
